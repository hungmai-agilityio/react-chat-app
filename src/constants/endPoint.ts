export const URL = {
  HOME: '',
  SIGN_IN: '/sign-in',
  SIGN_UP: '/sign-up'
}

export const END_POINT = {
  USER: '/users',
  PROFILE: '/profiles',
  CHAT: 'chats',
  MESSAGE: 'messages'
}