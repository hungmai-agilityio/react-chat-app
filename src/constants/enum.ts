enum SIZE {
  MINI = 'extra-small',
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large'
}

enum TYPE {
  PRIMARY = 'primary',
  SECOND = 'second',
  TERTIARY = 'tertiary',
  FILE = 'file',
  TEXT = 'text',
  PASSWORD = 'password'
}

enum POSITION {
  TOP_RIGHT = 'top-right',
  TOP_LEFT = 'top-left',
  BOT_RIGHT = 'bottom-right',
  BOT_LEFT = 'bottom-left'
}

export { SIZE, TYPE, POSITION };
