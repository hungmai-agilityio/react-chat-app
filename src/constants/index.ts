export * from '@/constants/enum';
export * from '@/constants/list';
export * from '@/constants/endPoint';
export * from '@/constants/regex';
export * from '@/constants/message'
