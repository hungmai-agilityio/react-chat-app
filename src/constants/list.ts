import { ITab } from '@/components/Tabs/Tab';
import { faEdit, faTrash } from '@fortawesome/free-solid-svg-icons';

export const USER_OPTIONS = [{ label: 'Profile', value: 'profile' }, { label: 'Log out', value: 'logout' }];

export const MESSAGE_OPTIONS = [
  { label: 'Edit', value: 'editMess', icon: faEdit },
  { label: 'Delete', value: 'deleteMess', icon: faTrash }
];

export const INFO_OPTIONS = [{ label: 'Chat info', value: 'infoChat' }];

export const LIST_TAB_USERS: Omit<ITab, 'index' | 'onClick'>[] = [
  {
    title: 'Chat',
    value: '1'
  },
  {
    title: 'Users',
    value: '2'
  }
];