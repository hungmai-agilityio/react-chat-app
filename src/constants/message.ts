export const MESSAGE_VALID = {
  INPUT_REQUIRED: '{field} is required',
  EMAIL_ERROR: 'Email is not valid',
  EMAIL_EXIST: 'Email has been registered, please try again!',
  PASSWORD_ERROR:
    'Minimum 6 characters, one uppercase, one lowercase, one number',
  CONFIRM_ERROR: 'Password confirmation do not match password',
};

export const MESSAGE_API = {
  SIGN_UP_ERROR: 'An error occurred. Please try again',
  SIGN_IN_ERROR: 'email or password is incorrect',
  UPDATE_PROFILE_SUCCESS: 'Update success',
  UPDATE_PROFILE_ERROR: 'Update error',
};