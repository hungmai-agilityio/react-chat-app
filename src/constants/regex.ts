const EMAIL_REGEX = /^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/;
const PASSWORD_REGEX = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{6,}$/;

export { EMAIL_REGEX, PASSWORD_REGEX };
