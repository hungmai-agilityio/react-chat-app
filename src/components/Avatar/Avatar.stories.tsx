import { Meta, StoryObj } from '@storybook/react';

// Constants
import { SIZE } from '@/constants';

// Components
import Avatar from '@/components/Avatar';

const meta: Meta<typeof Avatar> = {
  component: Avatar,
  parameters: {
    layout: 'centered'
  },
  tags: ['autodocs'],
  argTypes: {
    avatar: {
      description: 'Render image for img tags'
    },
    name: {
      description: 'Set name if there is no avatar or the avatar is corrupted'
    },
    size: {
      description: 'Set size optional for Avatar'
    },
    styles: {
      description: 'Add custom css'
    },
    circle: {
      description: 'Render Avatar with border radius full'
    },
    onClick: {
      description: 'Handle the event when clicking on the avatar'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Avatar>;

export const AvatarWithImageSmall: Story = {
  args: {
    avatar: 'assets/images/mock-image.jpg',
    size: SIZE.SMALL,
    name: 'Test User'
  }
};

export const AvatarWithoutImageSmall: Story = {
  args: {
    size: SIZE.SMALL,
    name: 'Test User'
  }
};

export const AvatarWithImageLarge: Story = {
  args: {
    avatar: 'assets/images/mock-image.jpg',
    size: SIZE.LARGE,
    name: 'Test User'
  }
};

export const AvatarWithoutImageLarge: Story = {
  args: {
    size: SIZE.LARGE,
    name: 'Test User'
  }
};

export const AvatarWithImageLargeCircle: Story = {
  args: {
    avatar: 'assets/images/mock-image.jpg',
    size: SIZE.LARGE,
    name: 'Test User',
    circle: true
  }
};

export const AvatarWithoutImageLargeCircle: Story = {
  args: {
    size: SIZE.LARGE,
    name: 'Test User',
    circle: true
  }
};
