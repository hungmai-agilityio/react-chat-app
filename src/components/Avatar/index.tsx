import { clsx } from 'clsx';

// Constants
import { SIZE } from '@/constants';

interface AvatarProps {
  avatar?: string;
  name?: string;
  size?: SIZE.SMALL | SIZE.LARGE;
  styles?: string;
  circle?: boolean;
  onClick?: () => void;
}

const Avatar = ({
  avatar,
  name,
  size,
  styles,
  circle,
  onClick
}: AvatarProps) => {
  const nameAvatar = name ? name.charAt(0).toUpperCase() : '';

  const sizeClasses = {
    [SIZE.SMALL]: 'w-avatar-sm h-avatar-sm text-lg',
    [SIZE.LARGE]: 'w-avatar-lg h-avatar-lg text-8xl'
  };

  const sizeClass = sizeClasses[size || SIZE.SMALL];

  if (!avatar) {
    return (
      <div
        data-testid="avatar"
        onClick={onClick}
        className={clsx(
          'text-primary bg-gray-300 flex items-center justify-center font-semibold',
          circle && 'rounded-full',
          styles,
          sizeClass
        )}
      >
        {nameAvatar}
      </div>
    );
  }

  return (
    <img
      data-testid="avatar"
      onClick={onClick}
      src={avatar}
      alt={name}
      className={clsx(
        'object-cover font-bold',
        circle && 'rounded-full',
        styles,
        sizeClass
      )}
    />
  );
};

export default Avatar;
