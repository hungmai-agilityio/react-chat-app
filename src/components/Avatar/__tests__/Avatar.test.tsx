import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

// Constants
import { SIZE } from '@/constants';

// Components
import Avatar from '@/components/Avatar';

describe('Avatar', () => {
  const propsDefault = {
    avatar: '',
    name: 'avatar',
    circle: false
  };

  it('renders avatar with image', () => {
    render(
      <Avatar
        {...propsDefault}
        avatar="assets/images/mock-image.jpg"
        size={SIZE.SMALL}
      />
    );

    const avatarImage = screen.getByAltText(propsDefault.name);
    expect(avatarImage).toBeInTheDocument();
  });

  it('renders avatar without image', () => {
    render(<Avatar {...propsDefault} avatar="" />);

    const avatarText = screen.getByText(
      propsDefault.name.charAt(0).toUpperCase()
    );
    expect(avatarText).toBeInTheDocument();
  });

  it('renders circle avatar', () => {
    render(
      <Avatar
        {...propsDefault}
        circle={true}
        avatar="assets/images/mock-image.jpg"
      />
    );

    const avatarElement = screen.getByAltText(propsDefault.name);
    expect(avatarElement).toHaveClass('rounded-full');
  });

  it('should render match to snapshot', () => {
    const { container } = render(<Avatar {...propsDefault} />);

    expect(container).toMatchSnapshot();
  });
});
