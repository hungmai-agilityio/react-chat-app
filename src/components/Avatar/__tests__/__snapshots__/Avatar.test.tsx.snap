// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render match to snapshot 1`] = `
<div>
  <div
    class="text-primary bg-gray-300 flex items-center justify-center font-semibold w-avatar-sm h-avatar-sm text-lg"
    data-testid="avatar"
  >
    A
  </div>
</div>
`;
